testmodule/fileparser.py

```diff
@@ -0,0 +1,101 @@
+class jobFile:
+    def __init__(self, file_path):
+        self.file_path = file_path
+        self.name = None
+        self.foldername = None
+        self.lines = None
+        self.headlines = []
+        self.programlines = []
+        self.separator = None
+
+        self.read_file()
+        self.save_name()
+        self.save_foldername()
+        self.rule_list()
+
+    def read_file(self):
+        """Class method to read the file and print the content."""
+
+        try:
+            with open(self.file_path, "r") as file:
+                self.lines = file.readlines()
+
+                for i, line in enumerate(self.lines):
+                    if line.startswith("NOP"):
+                        self.separator = i  # stores the index of NOP
+
+                        for c in range(i, len(self.lines)):
+                            self.programlines.append(
+                                self.lines[c]
+                            )  # add the programlines
+
+                        break
+                    else:
+                        self.headlines.append(
+                            line
+                        )  # add the lines before NOP into headlines
+
+        except FileNotFoundError:
+            print(f"File not found: {self.file_path}")
```

Comment:
When "catching" an exception I prefer to `raise` the exception instead of printing it out

This will give the more debugging information about the error